var/spack/repos/builtin/packages/py-dask/package.py

```diff
@@ -138,24 +145,37 @@ class PyDask(PythonPackage):
     # Requirements for dask.distributed
     depends_on("py-dill", type=("build", "run"), when="@:0.7.5 +distributed")
     depends_on("py-pyzmq", type=("build", "run"), when="@:0.7.5 +distributed")
-    depends_on("py-distributed@:2021.8.0", type=("build", "run"), when="@0.8.2: +distributed")
-    depends_on("py-distributed@1.9:2021.8.0", type=("build", "run"), when="@0.9.0: +distributed")
-    depends_on("py-distributed@1.10:2021.8.0", type=("build", "run"), when="@0.10.0: +distributed")
-    depends_on("py-distributed@1.14:2021.8.0", type=("build", "run"), when="@0.12.0: +distributed")
-    depends_on("py-distributed@1.15:2021.8.0", type=("build", "run"), when="@0.13.0: +distributed")
-    depends_on("py-distributed@1.16:2021.8.0", type=("build", "run"), when="@0.14.1: +distributed")
-    depends_on("py-distributed@1.20:2021.8.0", type=("build", "run"), when="@0.16.0: +distributed")
-    depends_on("py-distributed@1.21:2021.8.0", type=("build", "run"), when="@0.17.0: +distributed")
-    depends_on("py-distributed@1.22:2021.8.0", type=("build", "run"), when="@0.18.0: +distributed")
-    depends_on("py-distributed@2.0:2021.8.0", type=("build", "run"), when="@2.0.0: +distributed")
+    depends_on("py-distributed@:2021.8.0", type=("build", "run"), when="@0.8.2:0.8 +distributed")
+    depends_on("py-distributed@1.9:2021.8.0", type=("build", "run"), when="@0.9 +distributed")
+    depends_on("py-distributed@1.10:2021.8.0", type=("build", "run"), when="@0.10 +distributed")
+    depends_on("py-distributed@1.14:2021.8.0", type=("build", "run"), when="@0.12 +distributed")
+    depends_on("py-distributed@1.15:2021.8.0", type=("build", "run"), when="@0.13 +distributed")
     depends_on(
-        "py-distributed@2020.12.0:2021.8.0", type=("build", "run"), when="@2020.12.0: +distributed"
+        "py-distributed@1.16:2021.8.0", type=("build", "run"), when="@0.14.1:0.14 +distributed"
     )
-    depends_on("py-distributed@2021.6.2", type=("build", "run"), when="@2021.6.2 +distributed")
+    depends_on("py-distributed@1.20:2021.8.0", type=("build", "run"), when="@0.16 +distributed")
+    depends_on("py-distributed@1.21:2021.8.0", type=("build", "run"), when="@0.17 +distributed")
+    depends_on(
+        "py-distributed@1.22:2021.8.0", type=("build", "run"), when="@0.18.0:1 +distributed"
+    )
+    depends_on(
+        "py-distributed@2.0:2021.8.0", type=("build", "run"), when="@2.0.0:2020.11 +distributed"
+    )
+    depends_on(
+        "py-distributed@2020.12.0:2021.8.0",
+        type=("build", "run"),
+        when="@2020.12.0:2021.6.1 +distributed",
+    )
+    depends_on(
+        "py-distributed@2021.6.2", type=("build", "run"), when="@2021.6.2:2022.10.1 +distributed"
+    )
+    depends_on("py-distributed@2022.10.2", type=("build", "run"), when="@2022.10.2: +distributed")
```

Comment:
```suggestion
        "py-distributed@2021.6.2", type=("build", "run"), when="@2021.6.2 +distributed"
    )
    depends_on("py-distributed@2022.10.2", type=("build", "run"), when="@2022.10.2 +distributed")
```
These feel like matching version deps.